ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -835,9 +850,33 @@ static PrjFS_Result HandleHydrateFileRequest(const MessageHeader* request, const
     return result;
 }
 
+static PrjFS_Result HandleNewFileInRootNotification(
+    const MessageHeader* request,
+    const char* path,
+    const char* fullPath,
+    bool isDirectory,
+    PrjFS_NotificationType notificationType)
+{
+    // Whenever a new file shows up in the root, we need to check if its ancestor
+    // directories are flagged as in root.  If they are not, flag them as in root and
+    // notify the provider
+    FindNewFoldersInRootAndNotifyProvider(request, path);
+    
+    // TODO(Mac): Handle SetBitInFileFlags failures
+    SetBitInFileFlags(fullPath, FileFlags_IsInVirtualizationRoot, true);
```

Comment:
Same here, though this one is not new in this PR. Should we notify first, and set the bit second?

Reply:
> make sure we cover this scenario in the follow up robustness issue.

I filed #391 for this